### DISCUSSION TITLE
How can I integrate Mapbox's Mobile SDK's in-app navigation into my RN app?

### DISCUSSION THREAD

**Discussion by sauhardar** [rnmapbox/maps#1217] [Q&A]
I am building an app using this package in react native. Given this package doesn't support [in-app navigation](https://www.mapbox.com/navigation), I'm guessing I'll have to go down to the native level to be able to support such a feature. I don't have any experience writing native code on a react native app, so my question is whether this is even possible. If so, does anyone have any pointers on how I can get started?

Thanks again!

**Comment by sauhardar** [comment]
@ferdicus , any ideas?

**Reply by ferdicus** [reply]
So, two options:
1) you write your own react-native wrapper around mapboxes navigation sdk and use that - I guess it would only work with mapbox
2) you chose any service that has a routing api (HERE/ openstreetmap/ idk, there are probably a few out there), that also provide a tile service: render their tiles in our mapbox lib/ use there routing information to draw the route and so on

Unfortunately, there is no easy solution out of the box for this.
